src/d3d8/d3d8_device.cpp

```diff
@@ -926,6 +926,31 @@ namespace dxvk {
   }
 
   HRESULT STDMETHODCALLTYPE D3D8Device::SetViewport(const D3DVIEWPORT8* pViewport) {
+    if (likely(pViewport != nullptr)) {
+      HRESULT res;
+
+      // get the D3D9 render target if we haven't cached it yet
+      if (unlikely(m_renderTarget == nullptr)) {
+        Com<d3d9::IDirect3DSurface9> pRT9 = nullptr;
```

Comment:
It would be nicer if you moved that into a little helper function. As far as I can tell, we have that same code in a bunch of places.